src/mon/FSCommands.cc

```diff
@@ -273,6 +273,9 @@ class FsNewHandler : public FileSystemCommandHandler
     mon->osdmon()->do_set_pool_opt(metadata,
 				   pool_opts_t::PG_NUM_MIN,
 				   static_cast<int64_t>(16));
+    mon->osdmon()->do_set_pool_opt(metadata,
+				   pool_opts_t::PG_NUM_MAX,
+				   static_cast<int64_t>(16));
```

Comment:
Setting both the min and max to 16 effectively means pg_num cannot be tuned at all.  I think this bit should be dropped.

Reply:
if we don't set this for metadata pools they'll end up doing pg merging later, isn't that still expensive in bluestore?

the other place to set this if we need it for metadata pools is rgw - https://github.com/ceph/ceph/blob/ca614897822bef0540d89ef6c4100480b813ebc8/src/rgw/rgw_tools.cc#L78-L88

Reply:
We could set a max, but it should be more than the min :).  That raises the question of what it should be, though...  16x the min maybe (256)?

Reply:
@batrick @cbodley any opinions on max pgs for metadata pools?

Reply:
Is the pg_autoscaler not working well at scaling metadata pools? Thought Sage had that pretty well figured out?

Reply:
we've changed the algorithm to use the full pg budget of the cluster from the start, so you get full parallelism out of the box.

This PR is allowing us to constrain how many pgs a pool gets, so pools that do not need that kind of parallelism don't take up too much of the pg budget. The devicehealth pool is one example. The others we were thinking of adding a max to are the cephfs and rgw metadata pools.

Reply:
I see. Would it not be better to mark pools somehow that the extra parallelism is not needed?